tests/e2e/cmux_test.go

```diff
@@ -168,12 +170,21 @@ func validateGrpcgatewayRangeReponse(respData []byte) error {
 	return json.Unmarshal(respData, &resp)
 }
 
-func fetchMetrics(endpoint string, httpVersion string, connType e2e.ClientConnType) error {
-	req := e2e.CURLReq{Endpoint: "/metrics", Timeout: 5, HttpVersion: httpVersion}
-	respData, err := curl(endpoint, "GET", req, connType)
-	if err != nil {
+func fetchMetrics(t *testing.T, endpoint string, httpVersion string, connType e2e.ClientConnType) error {
+	tmpDir := t.TempDir()
+	metricFile := filepath.Join(tmpDir, "metrics")
+
+	req := e2e.CURLReq{Endpoint: "/metrics", Timeout: 5, HttpVersion: httpVersion, Output: metricFile}
+	if _, err := curl(endpoint, "GET", req, connType); err != nil {
 		return err
 	}
+
+	rawData, err := os.ReadFile(metricFile)
```

Comment:
Any reason that `curl` function shouldn't always do that?

Reply:
Oh, it would need access to `t` which would require a bigger PR that updates all curl calls.

Reply:
It's one of reasons. Other reason is that most of output are just few lines and we can just flush it into console. It will be easy to debug.